src/snabbdom.ts

```diff
@@ -1,12 +1,13 @@
 /* global module, document, Node */
 import {Module} from './modules/module';
-import {Hooks} from './hooks';
-import vnode, {VNode, VNodeData, Key} from './vnode';
+import vnode, {VNode} from './vnode';
 import * as is from './is';
 import htmlDomApi, {DOMAPI} from './htmldomapi';
 
+type NonUndefined<T> = T extends undefined ? never : T;
```

Comment:
So, TypeScript has a built-in type, `NonNullable<T>`. Are you aware of it? If you are, why did you choose to write this (admittedly more specific) type, instead of using it, please?

Reply:
I did it this way because `NonNullable` removes both `undefined` and `null` but the code in `isUndef` where the type is used only checks for `undefined`. Maybe we could get away with using `NonNullable` but I thought it would be best to have the type reflect exactly what the code actually does.